### PR TITLE
refactor: use for-await-of in updateDebt.js

### LINKED ISSUE #2258: convert `updateDebt.js` to use a for-await-of loop
## What is the Problem Being Solved?

Dean suggested:
https://github.com/Agoric/agoric-sdk/pull/2144/files#r562357092
When I do this naively, something breaks many tests in a way that isn't obvious, so splitting it into a separate task.

## Description of the Design

replace `addToDebtWhenNotified` with a for-await-of loop.

## Security Considerations

None. This shouldn't change the semantics

## Test Plan

There seem to be sufficient tests in this area.

### PR DESCRIPTION
**Pull request by Chris-Hibbert**
replaced a complex pile of iteration handling with a for-await-of loop.

Uncovered a bug in initializing the debt calculator: it would have
charged interest from the beginning of time.

It's not at all clear to me what the loan contract should do in case of failure in adding debt.

closes: #2258

### CODE REVIEW

File: packages/zoe/src/contracts/loan/updateDebt.js
Comment (dtribble) [inline packages/zoe/src/contracts/loan/updateDebt.js]: We really need to rename this helper method.

File: packages/zoe/src/contracts/loan/updateDebt.js
Comment (dtribble) [inline packages/zoe/src/contracts/loan/updateDebt.js]: why is this made async?  It completes synchronously.

File: packages/zoe/test/unitTests/contracts/loan/test-borrow.js
Comment (dtribble) [inline packages/zoe/test/unitTests/contracts/loan/test-borrow.js:317]: what causes the increment?  The first notification is always in a later turn?

**Review by dtribble — CHANGES_REQUESTED** [review]
Minor change request. So LGTM but I didn't want it to automatically merge until you looked at it.

File: packages/zoe/src/contracts/loan/updateDebt.js
Comment (katelynsills) [inline packages/zoe/src/contracts/loan/updateDebt.js:40]: You said there was a bug where we calculated from the start of time, but in initialization in what was line 96 `lastCalculationTimestamp` is set to `value`, so I don't think that's true. I like this version better, though.

**Review by katelynsills — CHANGES_REQUESTED** [review]
I agree with Dean's question of why the counts were incremented in the tests. Is that an effect of the code change? Or something that was desirable in the tests?

**Review by katelynsills — APPROVED** [review]
LGTM!

File: packages/zoe/src/contracts/loan/updateDebt.js
Comment (erights) [reply packages/zoe/src/contracts/loan/updateDebt.js]: See #2513

**Review by erights — CHANGES_REQUESTED** [review]

**Comment by erights** [comment]
See also #2514

File: packages/zoe/src/contracts/loan/updateDebt.js
Comment (erights) [inline packages/zoe/src/contracts/loan/updateDebt.js]: Sorry I forgot to do that in my PR-on-PR. But please do that before merging
```suggestion
  observeNotifier,
```

File: packages/zoe/src/contracts/loan/updateDebt.js
Comment (erights) [inline packages/zoe/src/contracts/loan/updateDebt.js]:
```suggestion
  observeNotifier(periodNotifier, periodObserver).catch(reason => {
```

**Review by katelynsills — APPROVED** [review]
LGTM!

File: packages/zoe/src/contracts/loan/updateDebt.js
Comment (erights) [reply packages/zoe/src/contracts/loan/updateDebt.js]: No longer appears. Resolving.

File: packages/zoe/src/contracts/loan/updateDebt.js
Comment (erights) [reply packages/zoe/src/contracts/loan/updateDebt.js]: I see it is done. Resolving.

File: packages/zoe/src/contracts/loan/updateDebt.js
Comment (erights) [reply packages/zoe/src/contracts/loan/updateDebt.js]: Resolving.

**Comment by Chris-Hibbert** [comment]
@erights,   github continues to claim that this PR is blocked on your requested changes. It looks like MarkM resolved all or most of the individual comments. Does that resolve all your issues?   

@dtribble,   would you tell github you've dropped your block? (Does github make that easy?)

**Review by erights — APPROVED** [review]
All my concerns were addressed.

**Comment by erights** [comment]
> @erights, github continues to claim that this PR is blocked on your requested changes. It looks like MarkM resolved all or most of the individual comments. Does that resolve all your issues?
> 
> @dtribble, would you tell github you've dropped your block? (Does github make that easy?)

I just submitted a new review checking the "Approve" radio button. Worked.

**Comment by Chris-Hibbert** [comment]
Dean [approved](https://github.com/Agoric/agoric-sdk/pull/2488#pullrequestreview-594700428). merging

**Pull request merged**
